src/Editor/main.cpp

```diff
@@ -158,9 +158,9 @@ int main() {
     // Shut down and cleanup.
     ImGuiImplementation::Shutdown();
     delete editor;
-    Hymn().world.Clear();
     
     Managers().ShutDown();
+    Hymn().world.Clear();
```

Comment:
This is not valid. Managers().ShutDown() deletes all the managers, but Hymn relies on the managers to clear its components in its Clear() method.

Reply:
The problem I was solving was actually solved by something else (clearing trigger components). The original problem was that component dependencies could not be handled in the destructor of `TriggerManager` because the owning entity and the component storage were released. This was in fact solved by simply calling `TriggerManager::ClearKilledComponents`, after which the destructor has nothing to do. I'll revert this change and replace the destructor contents with an assertion of this for future reference. Nice catch.